### PR TITLE
[MU4] Parts dialog

### PR DESCRIPTION
**Pull request by RomanPudashkin**
parts_dialog2

### CODE REVIEW

File: framework/uicomponents/qml/MuseScore/UiComponents/FlatButton.qml [hunk 1]
```diff
@@ -54,6 +54,8 @@ FocusableItem {
             id: buttonIcon
 
             anchors.horizontalCenter: parent.horizontalCenter
+
+            font.pixelSize: root.iconPixelSize
```
Comment (vpereverzev) [inline framework/uicomponents/qml/MuseScore/UiComponents/FlatButton.qml:58]: What is the motivation of this change? I believe it should depend on ui.theme

File: libmscore/score.cpp [hunk 2]
```diff
@@ -392,24 +392,25 @@ Score::~Score()
 
 Score* Score::clone()
 {
```
Comment (vpereverzev) [inline libmscore/score.cpp:394]: Could you explain literallly in few sentences what was wrong with the previous implementation?

File: libmscore/score.cpp [hunk 3]
```diff
@@ -392,24 +392,25 @@ Score::~Score()
...
+    Score* copy = new Score(masterScore());
+    excerpt.setPartScore(copy);
+    copy->_style = _style;
+    masterScore()->addExcerpt(&excerpt);
+    Excerpt::createExcerpt(&excerpt);
+    masterScore()->removeExcerpt(&excerpt);
+    excerpt.setPartScore(nullptr);
 
-    score->addLayoutFlags(LayoutFlag::FIX_PITCH_VELO);
-    score->doLayout();
```
Comment (vpereverzev) [inline libmscore/score.cpp:411]: I'm wondering if whether we still need this

File: libmscore/staff.h [hunk 4]
```diff
@@ -279,6 +279,7 @@ class Staff final : public Element
     void setPlaybackVoice(int voice, bool val) { _playbackVoice[voice] = val; }
 
     std::array<bool, VOICES> visibilityVoices() const { return _visibilityVoices; }
+    bool voiceVisible(int voice) const { return _visibilityVoices[voice]; }
```
Comment (vpereverzev) [inline libmscore/staff.h:282]: I'd propose "isVoiceVisible" name

Comment (vpereverzev) [reply libmscore/staff.h:282]: also we need to gradually move implementation of methods into .cpp files

File: mu4/context/internal/globalcontext.cpp [hunk 5]
```diff
@@ -79,6 +79,11 @@ void GlobalContext::setCurrentNotation(const INotationPtr& notation)
     }
 
     m_currentNotation = notation;
+
+    if (m_currentNotation) {
+        m_currentNotation->setOpened(true);
```
Comment (vpereverzev) [inline mu4/context/internal/globalcontext.cpp:84]: The question arises, where will it become false?

File: libmscore/score.cpp (hunk 2 above)
Comment (RomanPudashkin) [reply libmscore/score.cpp:394]: The previous implementation was fully copied from MasterScore::clone. It created MasterScore instance and returned pointer to it. Hence Notation stored pointer to MasterScore and used overrided methods from MasterScore, not from Score

File: framework/uicomponents/qml/MuseScore/UiComponents/FlatButton.qml (hunk 1 above)
Comment (RomanPudashkin) [reply framework/uicomponents/qml/MuseScore/UiComponents/FlatButton.qml:58]: You added this property by yourself, but it not worked :)
In some cases need to change the size of an icon. In my case, this is X button. Icon of this button is smaller than other icons
изображение

File: mu4/context/internal/globalcontext.cpp (hunk 5 above)
Comment (RomanPudashkin) [reply mu4/context/internal/globalcontext.cpp:84]: After closing notation via X button in this panel
изображение

File: libmscore/score.cpp (hunk 3 above)
Comment (RomanPudashkin) [reply libmscore/score.cpp:411]: They are called from Excerpt::createExcerpt

File: libmscore/staff.h (hunk 4 above)
Comment (RomanPudashkin) [reply libmscore/staff.h]: fixed

File: mu4/notation/internal/excerptnotation.cpp [hunk 6]
```diff
@@ -41,7 +45,35 @@ ExcerptNotation::~ExcerptNotation()
...
+void ExcerptNotation::setExcerpt(Ms::Excerpt* excerpt)
+{
+    m_excerpt = excerpt;
+    setScore(m_excerpt->partScore());
+    setMetaInfo(m_metaInfo);
+}
+
+Meta ExcerptNotation::metaInfo() const
+{
+    return isInited() ? Notation::metaInfo() : m_metaInfo;
```
Comment (Eism) [inline mu4/notation/internal/excerptnotation.cpp:57]: Strange condition. If initialized then should we return local data?

Comment (RomanPudashkin) [reply mu4/notation/internal/excerptnotation.cpp:57]: If notation was inited with default constructor then it has no score. We use local data in this case

**Review by Eism — APPROVED** [review]
